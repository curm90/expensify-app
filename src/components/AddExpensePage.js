import React from 'react';

const CreateExpensePage = () => (
  <div>
    <h1>Create expense page</h1>
  </div>
);

export default CreateExpensePage;
