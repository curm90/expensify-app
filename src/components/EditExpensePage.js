import React from 'react';

const EditExpensePage = () => (
  <div>
    <h1>Edit expense Page</h1>
  </div>
);

export default EditExpensePage;
