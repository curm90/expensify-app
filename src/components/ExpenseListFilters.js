import React from 'react';
import { connect } from 'react-redux';
import { DateRangePicker } from 'react-dates';
import {
  setTextFilter,
  sortByDate,
  sortByAmount,
  setStartDate,
  setEndDate,
} from '../actions/filters';

class ExpenseListFilters extends React.Component {
  state = {
    calenderFocused: null,
  };

  onDatesChange = ({ startDate, endDate }) => {
    this.props.dispatch(setStartDate(startDate));
    this.props.dispatch(setEndDate(endDate));
  };

  onFocusChange = (calenderFocused) => {
    this.setState(() => ({ calenderFocused }));
  };

  onTextChange = (e) => {
    this.props.dispatch(setTextFilter(e.target.value));
  };

  onSortChange = (e) => {
    if (e.target.value === 'date') {
      this.props.dispatch(sortByDate());
    } else if (e.target.value === 'amount') {
      this.props.dispatch(sortByAmount());
    }
  };

  render() {
    return (
      <div>
        <input
          type='text'
          value={this.props.filters.text}
          onChange={this.onTextChange}
        />
        <select
          value={this.props.filters.sortBy}
          onChange={this.onSortChange}
        >
          <option value='date'>Date</option>
          <option value='amount'>Amount</option>
        </select>

        <DateRangePicker
          startDate={this.props.filters.startDate}
          endDate={this.props.filters.endDate}
          onDatesChange={this.onDatesChange}
          focusedInput={this.state.calenderFocused}
          onFocusChange={this.onFocusChange}
          numberOfMonths={1}
          isOutsideRange={() => false}
          showClearDates={true}
        />
      </div>
    );
  }
}

const mapStateToProps = (state) => {
  return {
    filters: state.filters,
  };
};

export default connect(mapStateToProps)(ExpenseListFilters);
