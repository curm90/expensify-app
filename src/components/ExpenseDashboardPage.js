import React from 'react';
import ExpenseList from './ExpenseList';
import ExpenseListFilters from './ExpenseListFilters';
import ExpensesSummary from './ExpensesSummary';

const Homepage = () => (
  <div>
    <h1>Homepage</h1>
    <ExpensesSummary />
    <ExpenseListFilters />
    <ExpenseList />
  </div>
);

export default Homepage;
