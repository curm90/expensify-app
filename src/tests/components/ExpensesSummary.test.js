import React from 'react';
import { shallow } from '../setupTests';
import toJSON from 'enzyme-to-json';
import { ExpensesSummary } from '../../components/ExpensesSummary';

it('should correctly render ExpensesSummary with 1 expense', () => {
  const wrapper = shallow(
    <ExpensesSummary expenseCount={1} expensesTotal={235} />
  );
  expect(toJSON(wrapper)).toMatchSnapshot();
});

it('should correctly render ExpensesSummary with mulitple expenses', () => {
  const wrapper = shallow(
    <ExpensesSummary expenseCount={23} expensesTotal={132324343223} />
  );
  expect(toJSON(wrapper)).toMatchSnapshot();
});
