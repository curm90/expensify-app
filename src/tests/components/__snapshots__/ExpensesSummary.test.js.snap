// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render ExpensesSummary with 1 expense 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expense
     totalling
     
    $2.35
  </h1>
</div>
`;

exports[`should correctly render ExpensesSummary with mulitple expenses 1`] = `
<div>
  <h1>
    Viewing 
    23
     
    expenses
     totalling
     
    $1,323,243,432.23
  </h1>
</div>
`;
